ofrak_core/ofrak/gui/server.py

```diff
@@ -983,15 +982,14 @@ async def get_tags_and_num_components(self, request: Request):
     @exceptions_to_http(SerializedError)
     async def search_data(self, request: Request) -> Response:
         resource: Resource = await self._get_resource_for_request(request)
-        mode = request.query.get("type")
-        regex = request.query.get("regex") == "true"
-        case_ignore = request.query.get("caseignore") == "true"
-
+        body = await request.json()
+        mode = body.get("type")
+        regex = body.get("regex") == "true"
+        case_ignore = body.get("caseignore") == "true"
```

Comment:
JSON has a boolean type. These should be serialized as booleans, no? I suspect that either the values are being serialized wrong on the front end, or these checks will never be true.